torch/_dynamo/eval_frame.py

```diff
@@ -262,19 +264,125 @@ def innermost_fn(fn):
     return unaltered_fn
 
 
+dynamo_guarded_config_ignorelist = {
+    "log_file_name",
+    "verbose",
+    "verify_correctness",  # will not affect model, will raise RuntimeError
+    # (no silent change to compilation behaviour)
+    "cache_size_limit",
+    "accumulated_cache_size_limit",
+    "print_specializations",
+    "replay_record_enabled",
+    "cprofile",  # only wraps _compile, not graph
+    "repro_after",
+    "repro_level",
+    "repro_forward_only",
+    "repro_tolerance",
+    "same_two_models_use_fp64",
+    "error_on_recompile",  # safe because: will throw error
+    "report_guard_failures",
+    "report_all_guard_failures",
+    "base_dir",  # used for minifying / logging
+    "translation_validation",
+    "translation_validation_timeout",
+    "translation_validation_no_bisect",
+    "DEBUG_DIR_VAR_NAME",
+    "debug_dir_root",
+}
+
+dynamo_guarded_config_string_serialization_list = {
+    # "is_fbcode",  # function def (removed by ConfigModule visit)
+    "skipfiles_inline_module_allowlist",  # cannot pickle module (unused)
+    "constant_functions",  # PyCapsules (i.e. function ptrs)
+    "traceable_tensor_subclasses",  # Cannot pickle local object
```

Comment:
this is a bit unfortunate because these do materially affect compilation. One saving grace is that the "hash" here only needs to be valid within a single Python process, so we could potentially do something like just directly hash on ids

Reply:
Is it ok to use their `__repr__` as suggested elsewhere? It's not ideal but likely to do the job in most cases... And much more human readable and deterministic than using the ids.

Reply:
resolved in https://github.com/pytorch/pytorch/pull/111299